Problem running build_all_ios.sh

Mac OS X: El Capitan (10.11.4)

I am trying to build Tensorflow for iOS and am following the "Building all at once" instructions found [here](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/makefile/README.md).  When running the build_all_ios.sh file, I am getting the following errors:

In file included from /Users/mw/Tensorflow_gitclone/tensorflow/tensorflow/contrib/makefile/downloads/eigen-eigen-802d984ade26/unsupported/Eigen/CXX11/../../../Eigen/Core:355:
/Users/mw/Tensorflow_gitclone/tensorflow/tensorflow/contrib/makefile/downloads/eigen-eigen-802d984ade26/unsupported/Eigen/CXX11/../../../Eigen/src/Core/arch/NEON/Complex.h:18:35: error: 
      statement expression not allowed at file scope
static uint32x4_t p4ui_CONJ_XOR = vld1q_u32( conj_XOR_DATA );
                                  
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../lib/clang/7.3.0/include/arm_neon.h:7609:39: note: 
      expanded from macro 'vld1q_u32'
#define vld1q_u32(__p0) __extension__ ({ \
                                      
In file included from tensorflow/core/kernels/xent_op.cc:20:
In file included from ./tensorflow/core/kernels/xent_op.h:20:
In file included from ./third_party/eigen3/unsupported/Eigen/CXX11/Tensor:1:
In file included from /Users/mw/Tensorflow_gitclone/tensorflow/tensorflow/contrib/makefile/downloads/eigen-eigen-802d984ade26/unsupported/Eigen/CXX11/Tensor:14:
In file included from /Users/mw/Tensorflow_gitclone/tensorflow/tensorflow/contrib/makefile/downloads/eigen-eigen-802d984ade26/unsupported/Eigen/CXX11/../../../Eigen/Core:355:
/Users/mw/Tensorflow_gitclone/tensorflow/tensorflow/contrib/makefile/downloads/eigen-eigen-802d984ade26/unsupported/Eigen/CXX11/../../../Eigen/src/Core/arch/NEON/Complex.h:19:35: error: 
      statement expression not allowed at file scope
static uint32x2_t p2ui_CONJ_XOR = vld1_u32( conj_XOR_DATA );
                                  
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../lib/clang/7.3.0/include/arm_neon.h:7759:38: note: 
      expanded from macro 'vld1_u32'
#define vld1_u32(__p0) __extension__ ({ \
                                     
2 errors generated.
make: *** [/Users/mw/Tensorflow_gitclone/tensorflow/tensorflow/contrib/makefile/gen/obj/tensorflow/core/kernels/xent_op.o] Error 1
+ '[' 2 -ne 0 ']'
+ echo 'armv7 compilation failed.'
armv7 compilation failed.
+ exit 1
MW:makefile mw$ ./build_all_ios.sh


I have been able to overcome many issues up to this one but am now stuck.  Any suggestions?

--- a/tensorflow/contrib/makefile/compile_ios_protobuf.sh
+++ b/tensorflow/contrib/makefile/compile_ios_protobuf.sh
@@ -1,4 +1,4 @@
-#!/bin/bash -x
+#!/bin/bash -x -e
 # Copyright 2015 The TensorFlow Authors. All Rights Reserved.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
@@ -14,6 +14,9 @@
 # limitations under the License.
 # ==============================================================================
 # Builds protobuf 3 for iOS.
+
+SCRIPT_DIR=$(dirname $0)
+source "${SCRIPT_DIR}/build_helper.subr"
 
 cd tensorflow/contrib/makefile
 

--- a/tensorflow/contrib/makefile/compile_ios_tensorflow.sh
+++ b/tensorflow/contrib/makefile/compile_ios_tensorflow.sh
@@ -28,6 +28,18 @@ GENDIR=tensorflow/contrib/makefile/gen/
 LIBDIR=${GENDIR}lib
 LIB_PREFIX=libtensorflow-core
 
+# FIXME(petewarden) - Some new code in Eigen triggers a clang bug, so work
+# around it by patching the source.
+sed -e 's#static uint32x4_t p4ui_CONJ_XOR = vld1q_u32( conj_XOR_DATA );#static uint32x4_t p4ui_CONJ_XOR; // = vld1q_u32( conj_XOR_DATA ); - Removed by script#' \
+-i '' \
+tensorflow/contrib/makefile/downloads/eigen-latest/eigen/src/Core/arch/NEON/Complex.h
+sed -e 's#static uint32x2_t p2ui_CONJ_XOR = vld1_u32( conj_XOR_DATA );#static uint32x2_t p2ui_CONJ_XOR;// = vld1_u32( conj_XOR_DATA ); - Removed by scripts#' \
+-i '' \
+tensorflow/contrib/makefile/downloads/eigen-latest/eigen/src/Core/arch/NEON/Complex.h
+sed -e 's#static uint64x2_t p2ul_CONJ_XOR = vld1q_u64( p2ul_conj_XOR_DATA );#static uint64x2_t p2ul_CONJ_XOR;// = vld1q_u64( p2ul_conj_XOR_DATA ); - Removed by script#' \
+-i '' \
+tensorflow/contrib/makefile/downloads/eigen-latest/eigen/src/Core/arch/NEON/Complex.h
+
 make -f tensorflow/contrib/makefile/Makefile cleantarget
 make -f tensorflow/contrib/makefile/Makefile \
 TARGET=IOS IOS_ARCH=ARMV7 LIB_NAME=${LIB_PREFIX}-armv7.a OPTFLAGS="$1" $2 $3

--- a/tensorflow/contrib/makefile/tf_cc_files.txt
+++ b/tensorflow/contrib/makefile/tf_cc_files.txt
@@ -227,6 +227,7 @@ tensorflow/core/framework/graph_def_util.cc
 tensorflow/core/framework/function.cc
 tensorflow/core/framework/fake_input.cc
 tensorflow/core/framework/device_base.cc
+tensorflow/core/framework/common_shape_fns.cc
 tensorflow/core/framework/cancellation.cc
 tensorflow/core/framework/bfloat16.cc
 tensorflow/core/framework/attr_value_util.cc

